src/vmm/src/vstate.rs

```diff
@@ -774,18 +775,33 @@ impl Vcpu {
             // Unhandled exit of the other end.
             Err(_) => {
                 // Move to 'exited' state.
-                StateMachine::next(Self::exited)
+                self.exit(FC_EXIT_CODE_GENERIC_ERROR)
             }
         }
     }
 
-    // This is the main loop of the `Exited` state.
-    fn exited(&mut self) -> StateMachine<Self> {
+    // Transition to the exited state
+    fn exit(&mut self, exit_code: u8) -> StateMachine<Self> {
```

Comment:
nit: why is this a separate function and not just the `exited` state?

Reply:
Because it needs to receive the `exit_code` as a parameter. From what I understand a `StateFn` can't receive parameters.

Reply:
Personally I don't like it too much because exit_code is not generic state for the vCPU. It's just a code that has to be use once, right before exit. It would be really nice if we could send parameters to the state functions. Since it's not possible for the moment, I personally prefer to have the intermediate function `exit()` that receives the exit_code parameter.